neet/automata.py

```diff
@@ -0,0 +1,247 @@
+        :raises ValueError: if ``code`` is not in :math:`\{0,1,\ldots,255\}`
+        """
+        return self.__code
+
+    @code.setter
+    def code(self, code):
+        if not isinstance(code, int):
+            raise(TypeError("ECA code is not an int"))
+        if 255 < code or code < 0:
+            raise(ValueError("invalid ECA code"))
+        self.__code = code
+
+    @property
+    def boundary(self):
+        """
+        The boundary conditions of the elemenary cellular automaton
+
+        .. rubric:: Examples:
+
+        ::
+
+            >>> eca = ECA(30)
+            >>> eca.boundary
+            >>> eca.boundary = (0,1)
+            >>> eca.boundary
+            (0, 1)
+            >>> eca.boundary = None
+            >>> eca.boundary
+            >>> eca.boundary = [0,1]
+            Traceback (most recent call last):
+              File "<stdin>", line 1, in <module>
+              File "neet/automata.py", line 110, in boundary
+                raise(TypeError("ECA boundary are neither None nor a tuple"))
+            TypeError: ECA boundary are neither None nor a tuple
+
+        :type: ``None`` or tuple
+        :raises TypeError: if ``boundary`` is neither ``None`` or an instance of tuple
+        :raises ValueError: if ``boundary`` is a neither ``None`` or a pair of binary states
+        """
+        return self.__boundary
+
+    @boundary.setter
+    def boundary(self, boundary):
+        if boundary and not isinstance(boundary, tuple):
+            raise(TypeError("ECA boundary are neither None nor a tuple"))
+        if boundary:
+            if len(boundary) != 2:
+                raise(ValueError("invalid ECA boundary conditions"));
+            for x in boundary:
+                if x != 0 and x != 1:
+                    raise(ValueError("invalid ECA boundary value"))
+        self.__boundary = boundary
+
+    @classmethod
+    def check_lattice(self, lattice):
+        """
+        Check the validity of the provided lattice
+
+        .. rubric:: Examples:
+
+        ::
+
+            >>> ECA.check_lattice([0,0,0])
+            True
+            >>> ECA.check_lattice([])
+            Traceback (most recent call last):
+              File "<stdin>", line 1, in <module>
+              File "neet/automata.py", line 130, in check_lattice
+                raise(ValueError("lattice is too short"))
+            ValueError: lattice is empty
+
+            >>> ECA.check_lattice([0,0,2])
+            Traceback (most recent call last):
+              File "<stdin>", line 1, in <module>
+              File "neet/automata.py", line 135, in check_lattice
+                raise(ValueError(msg))
+            ValueError: invalid value "2" in lattice
+
+            >>> ECA.check_lattice(5)
+            Traceback (most recent call last):
+              File "<stdin>", line 1, in <module>
+              File "neet/automata.py", line 129, in check_lattice
+                if len(lattice) < 3:
```

Comment:
Commit 93f24ff fixes the incorrect docstring by removing the traceback.

You are correct that generally all a CA needs is lattice with at least one element. Commit 7a94d1c adds unit tests to reflect that the API supports lattice sizes down to length 1.